config/clients/ruby/template/lib/openfga/client/openfga_client.rb.mustache

```diff
@@ -0,0 +1,490 @@
+        end
+
+        # Check for duplicates
+        if correlation_ids.include?(correlation_id)
+          fail ArgumentError, "Duplicate correlation_id found: #{correlation_id}"
+        end
+        correlation_ids.add(correlation_id)
+      end
+
+      # If we have fewer checks than the batch limit, use simple approach
+      if checks.length <= max_batch_size
+        return execute_single_batch_check(checks, opts)
+      end
+
+      # Split checks into batches and process concurrently
+      check_batches = checks.each_slice(max_batch_size).to_a
+      all_results = process_batches_concurrently(check_batches, max_parallel_requests, opts)
+
+      # Merge all batch results
+      merged_results = {}
+      all_results.each do |batch_response|
+        if batch_response&.result
+          merged_results.merge!(batch_response.result)
+        end
+      end
+
+      # Return a BatchCheckResponse with merged results
+      BatchCheckResponse.new(result: merged_results)
+    end
+
+    # Read changes
+    # Reads the list of historical relationship tuple writes and deletes.
+    # @param type [String] :type Get the list of tuple changes that affect only this type
+    # @param start_time [String] :start_time The start time of the range to read changes from. This is a timestamp in ISO 8601 format.
+    # @param [Hash] opts the optional parameters
+    # @option opts [Integer] :page_size The number of pages to return in the request
+    # @option opts [String] :continuation_token The continuation token to use to get the next page of results. This will be empty if there are no more results.
+    # @option opts [String] :store_id The store ID to read changes from
+    def read_changes(type:, start_time:, opts: {})
+      fail ArgumentError, "Missing the required parameter 'type'" if type.nil?
+      fail ArgumentError, "Missing the required parameter 'start_time'" if start_time.nil?
+
+      @api_client.read_changes(store_id(opts), opts.merge(type:, start_time:))
+    end
+
+    # GET /stores/{store_id}/assertions/{authorization_model_id}
+    # Retrieve assertions for a specific store and authorization model
+    # @param [Hash] opts The optional parameters
+    # @return [GetAssertionsResponse]
+    def read_assertions(opts = {})
+      @api_client.read_assertions(store_id(opts), authorization_model_id(opts), opts)
+    end
+
+    # PUT /stores/{store_id}/assertions/{authorization_model_id}
+    # Update assertions for a specific store and authorization model
+    # @param assertions [Hash] The request body containing the assertions
+    # @param [Hash] opts The optional parameters
+    # @return [nil]
+    def write_assertions(assertions:, opts: {})
+      fail ArgumentError, "Missing the required parameter 'assertions'" if assertions.nil? || assertions.empty?
+
+      request_body = WriteAssertionsRequest.new(assertions:)
```

Comment:
Can we have this consistent with the other SDKs?

E.g. https://github.com/openfga/js-sdk/blob/b811c2d6cd95765ddd2d203121e45b9957cbcc7f/client.ts#L904C37-L914

Note: feel free to write your own models for the client - e.g. `ClientWriteAssertionsRequest`